internal/telemetry/configuration.go

```diff
@@ -23,6 +23,13 @@ type configuration struct {
 	seqID  uint64
 }
 
+func idOrEmpty(id string) string {
+	if id == EmptyID {
```

Comment:
What happens if `id` is a string only containing blank characters, like `"   "`? Is that possible? Also, provided that `EmptyID` is a constant string, isn't this function actually a no-op? In both branches we are returning the effective `id` value.